mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/task/SolveMailboxInconsistenciesService.java

```diff
@@ -0,0 +1,363 @@
+        Mono<Result> fix(Context context, CassandraMailboxDAO mailboxDAO, CassandraMailboxPathV2DAO pathV2DAO);
+    }
+
+    private static class NoInconsistency implements Inconsistency{
+        @Override
+        public Mono<Result> fix(Context context, CassandraMailboxDAO mailboxDAO, CassandraMailboxPathV2DAO pathV2DAO) {
+            return Mono.just(Result.COMPLETED);
+        }
+    }
+
+    /**
+     * The Mailbox is referenced in MailboxDAO but the corresponding
+     * reference is missing in MailboxPathDao.
+     *
+     * In order to solve this inconsistency, we can simple re-reference the mailboxPath.
+     */
+    private static class OrphanMailboxDAOEntry implements Inconsistency {
+        private final Mailbox mailbox;
+
+        private OrphanMailboxDAOEntry(Mailbox mailbox) {
+            this.mailbox = mailbox;
+        }
+
+        @Override
+        public Mono<Result> fix(Context context, CassandraMailboxDAO mailboxDAO, CassandraMailboxPathV2DAO pathV2DAO) {
+            return pathV2DAO.save(mailbox.generateAssociatedPath(), (CassandraId) mailbox.getMailboxId())
+                .map(success -> {
+                    if (success) {
+                        LOGGER.info("Fixing inconsistency for orphan mailbox {} - {}",
+                            mailbox.getMailboxId().serialize(),
+                            mailbox.generateAssociatedPath().asString());
+                        context.fixedInconsistencies.incrementAndGet();
+                        return Result.COMPLETED;
+                    } else {
+                        context.errors.incrementAndGet();
+                        LOGGER.warn("Fail fixing inconsistency for orphan mailbox {} - {}",
+                            mailbox.getMailboxId().serialize(),
+                            mailbox.generateAssociatedPath().asString());
+                        return Result.PARTIAL;
+                    }
+                });
+        }
+    }
+
+    /**
+     * The Mailbox is referenced in MailboxPathDao but the corresponding
+     * entry is missing in MailboxDao.
+     *
+     * CassandraIds are guaranteed to be unique, and are immutable once set to a mailbox.
+     *
+     * This this inconsistency arise if mailbox creation fails or upon partial deletes.
+     *
+     * In both case removing the dandling path registration solves the inconsistency
+     *
+     * In order to solve this inconsistency, we can simple re-reference the mailboxPath.
+     */
+    private static class OrphanMailboxPathDAOEntry implements Inconsistency {
+        private final CassandraIdAndPath pathRegistration;
+
+        private OrphanMailboxPathDAOEntry(CassandraIdAndPath pathRegistration) {
+            this.pathRegistration = pathRegistration;
+        }
+
+        @Override
+        public Mono<Result> fix(Context context, CassandraMailboxDAO mailboxDAO, CassandraMailboxPathV2DAO pathV2DAO) {
+            return pathV2DAO.delete(pathRegistration.getMailboxPath())
```

Comment:
I thought if the path exists, we create the mailbox, not deleting it... Did you mix between the cases or did I misunderstand?

Reply:
If the path exist but not the mailbox, then we miss the UID validity information.

Also, the 'create' did not succeeded, it seems logical to me that after solving the inconsistency we endup with no mailbox here.

The goal is not `to fix failed creates` but just ensure that they correct our data.

I believe the ticket description is wrong then.